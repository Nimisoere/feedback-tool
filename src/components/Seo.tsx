import Head from 'next/head'
import React from 'react'

type Props = {
  title: string;
  description?: string;
  keywords?: string[];
  noIndex?: boolean;
  noFollow?: boolean;
}

const Seo = ({ title, noFollow, noIndex, description,keywords }: Props) => {
  return (
    <Head>
      <title>{title}</title>
      <meta name="viewport" content="width=device-width, initial-scale=1" />
      {noIndex && <meta name="robots" content="noIndex" />}
      {noFollow && <meta name="robots" content="noFollow" />}
      <meta property="og:title" content={title} />
      <meta name="twitter:title" content={title} />
      {description && <meta name="description" content={description} />}
      {description && (
        <meta property="og:description" content={description} />
      )}
      {description && (
        <meta name="twitter:description" content={description} />
      )}
      <meta name="twitter:card" content="summary_large_image" />
      {keywords?.length && <meta name="keywords" content={keywords?.join(",")} />}
    </Head>
  )
}

export default Seo