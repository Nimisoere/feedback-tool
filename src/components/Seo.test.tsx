import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import Seo from "./Seo";

describe("Feedback Graph", () => {
  it("should match snapshot", async () => {
    const { getByText, container } = render(
      <Seo title="Test" description="test" keywords={["test-1", "test-2"]} noFollow noIndex />
    );
    expect(container).toMatchSnapshot();
  });
});
