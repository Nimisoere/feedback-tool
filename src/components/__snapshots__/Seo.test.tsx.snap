// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Graph should match snapshot 1`] = `<div />`;
